docs/recipes/amp-alertmanager-terraform.md

```diff
@@ -0,0 +1,141 @@
+
+```
+curl https://raw.githubusercontent.com/aws-observability/aws-otel-collector/main/examples/eks/aws-prometheus/prometheus-daemonset.yaml -o prometheus-daemonset.yaml
+```
+Then edit the template file, substituting the remote_write endpoint for your Amazon Managed Service for Prometheus workspace for YOUR_ENDPOINT and your Region for YOUR_REGION. Use the remote_write endpoint that is displayed in the Amazon Managed Service for Prometheus console when you look at your workspace details.
+You'll also need to change YOUR_ACCOUNT_ID in the service account section of the Kubernetes configuration to your AWS account ID.
+In this recipe, the ADOT Collector configuration uses an annotation (scrape=true) to tell which target endpoints to scrape. This allows the ADOT Collector to distinguish the sample app endpoint from kube-system endpoints in your cluster. You can remove this from the re-label configurations if you want to scrape a different sample app.
+5. Enter the following command to deploy the ADOT collector.
+```
+kubectl apply -f eks-prometheus-daemonset.yaml
+```
+
+### Deploying the Terraform module to configure Amazon Managed service for Prometheus workspace, recording rules & alert manager
+
+Now, we will  provision a Amazon Managed service for Prometheus workspace and will define an alerting rule that causes the Alert Manager to send a notification if a certain condition (defined in expr) holds true for a specified time period (for). Code in the Terraform language is stored in plain text files with the .tf file extension. There is also a JSON-based variant of the language that is named with the .tf.json file extension.
+
+```
+export SNS_TOPIC=<your SNS topic ARN>
+cat << EOF > main.tf
+provider "aws" {
+  profile    = "default"
+  region     = "$AWS_REGION"
+}
+
+resource "aws_prometheus_workspace" "amp-terraform-ws" {
+  alias = "amp-terraform-ws"
+}
+
+resource "aws_prometheus_rule_group_namespace" "amp-terraform-ws" {
+  name         = "rules"
+  workspace_id = aws_prometheus_workspace.amp-terraform-ws.id
+  data         = <<EOF
+groups:
+  - name: test
+    rules:
+    - record: metric:recording_rule
+      expr: rate(adot_test_counter0[5m])
+  - name: alert-test
+    rules:
+    - alert: metric:alerting_rule
+      expr: rate(adot_test_counter0[5m]) > 0.014
+      for: 5m    
+EOF
+}
+
+resource "aws_prometheus_alert_manager_definition" "amp-terraform-ws" {
+  workspace_id = aws_prometheus_workspace.amp-terraform-ws.id
+  definition   = <<EOF
+alertmanager_config: |
+  route:
+    receiver: 'default'
+  receivers:
+    - name: 'default'
+      sns_configs:
+      - topic_arn: $SNS_TOPIC
+        sigv4:
+          region: $AWS_REGION
+        attributes:
+          key: severity
+          value: SEV2
+EOF
+}
+```
+
+Once the file is created, execute the below commands to provision the resource using terraform:
```

Comment:
Turn above file into a proper resource (supporting file as per https://github.com/aws-observability/aws-o11y-recipes/#writing-recipes) and make sure to clean it up. It looks like there's an extra `EOF` in there?

Reply:
have created a file main.tf as per the direction